// angular-feather/esm2015/feather/feather.module.js
import * as tslib_1 from "tslib";
var FeatherModule_1;
import { NgModule, Optional } from '@angular/core';
import { FeatherComponent } from './feather.component';
import { Icons } from './icons.provider';
let FeatherModule = FeatherModule_1 = class FeatherModule {
    constructor(icons) {
        this.icons = icons;
        if (!this.icons) {
            throw new Error(`No icon provided. Make sure to use 'FeatherModule.pick({ ... })' when importing the module\n` +
                `Refer to documentation on https://github.com/michaelbazos/angular-feather`);
        }
    }
    static pick(icons) {
        return {
            ngModule: FeatherModule_1,
            providers: [
                { provide: Icons, multi: true, useValue: icons }
            ]
        };
    }
};
FeatherModule.ctorParameters = () => [
    { type: Icons, decorators: [{ type: Optional }] }
];
FeatherModule = FeatherModule_1 = tslib_1.__decorate([
    NgModule({
        declarations: [
            FeatherComponent
        ],
        exports: [
            FeatherComponent
        ]
    }),
    tslib_1.__param(0, Optional())
], FeatherModule);
export { FeatherModule };
